presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -605,6 +605,16 @@ std::unique_ptr<common::Filter> toFilter(
       return combineIntegerRanges(bigintFilters, nullAllowed);
     }
 
+    if (type->kind() == TypeKind::DATE) {
+      std::vector<std::unique_ptr<common::BigintRange>> dateFilters;
+      dateFilters.reserve(ranges.size());
+      for (const auto& range : ranges) {
+        dateFilters.emplace_back(
+            dateRangeToFilter(range, nullAllowed, exprConverter, type));
+      }
+      return combineIntegerRanges(dateFilters, nullAllowed);
```

Comment:
Do we require to do this integer range combination logic for date filters ? Ideally, date range is from 01-01-0001 to 12-31-9999, and the complex logic in combineIntegerRanges might be overkill for dates ?

Reply:
That is true, having a complicated combination logic is not required for date filters because, as you mentioned, it is a much smaller range. I suppose I just wanted to keep this similar to integer ranges and avoid implementing `combineDateRanges` function since `combineIntegerRanges` is a superset.

Would you like to have a separate `combineDateRanges` function for date ranges?

Reply:
Can the date ranges just be retained as originally specified without a combine function ? It should be fine to keep the original ranges just so I think.

Reply:
combineIntegerRanges is a misnomer in the sense its not really combining range filters when there is overlap. It is doing :
i) Creating a values filter if all the filters are checking a single value
ii) Creating NegatedBigIntRange filters if there is only a small big int interval is that is omitted by the range filters.
iii) Else its returning a MultiRange Filter with the individual ranges.

For dates specifically the negated filters logic will not get triggered. For the most part only MultiRangeFilter will be returned. So this function is not really useful for Date filters.

In general, I feel its better to return a BigIntMultiRange filter here and then add optimizations later as needed.

Reply:
I can follow up on streamlining the logic for Date type once this PR is merged if that works.

Reply:
@aditi-pandit thanks for the clarification.

Reply:
I will update, thanks!